lib/iris/experimental/ugrid/__init__.py

```diff
@@ -3274,6 +3283,134 @@ def context(self):
 PARSE_UGRID_ON_LOAD = ParseUGridOnLoad()
 
 
+def _meshes_from_cf(cf_reader):
+    """
+    Common behaviour for extracting meshes from a CFReader.
+
+    Simple now, but expected to increase in complexity as Mesh sharing develops.
+
+    """
+    # Mesh instances are shared between file phenomena.
+    # TODO: more sophisticated Mesh sharing between files.
+    # TODO: access external Mesh cache?
+    mesh_vars = cf_reader.cf_group.meshes
+    meshes = {
+        name: _build_mesh(cf_reader, var, cf_reader.filename)
+        for name, var in mesh_vars.items()
+    }
+    return meshes
+
+
+def load_mesh(uris, var_name=None):
+    """
+    Create a single :class:`Mesh` object from one or more NetCDF files.
+
+    Raises an error if more/less than one :class:`Mesh` is found.
+
+    Parameters
+    ----------
+    uris : str or iterable of str
+        One or more filenames/URI's. Any URI's must support OpenDAP.
+    var_name : str, optional
+        Only return a :class:`Mesh` if its var_name matches this value.
+
+    Returns
+    -------
+    :class:`Mesh`
+
+    """
+    meshes_result = load_meshes(uris, var_name)
+    result = [mesh for file in meshes_result.values() for mesh in file]
+    mesh_count = len(result)
+    if mesh_count != 1:
+        message = (
+            f"Expecting 1 mesh, but input file(s) produced: {mesh_count} ."
+        )
+        raise ConstraintMismatchError(message)
+    return result[0]
+
+
+def load_meshes(uris, var_name=None):
+    """
+    Create :class:`Mesh` objects from one or more NetCDF files.
+
+    Parameters
+    ----------
+    uris : str or iterable of str
+        One or more filenames/URI's. Any URI's must support OpenDAP.
+    var_name : str, optional
+        Only return a :class:`Mesh` if its var_name matches this value.
```

Comment:
```suggestion
        Only return :class:`Mesh`'s whose var_name matches this.
```
?maybe slightly clearer?

Reply:
8a981a7997008e3ec815a1f04d8a5b69cd19b995